libraries/Networking/src/NetworkInterface.cpp

```diff
@@ -0,0 +1,719 @@
+        if(!_interface_event_group){
+            log_e("Interface Event Group Create Failed!");
+            return false;
+        }
+        setStatusBits(_initial_bits);
+    }
+
+    if(_ip_ev_instance == NULL && esp_event_handler_instance_register(IP_EVENT, ESP_EVENT_ANY_ID, &_ip_event_cb, NULL, &_ip_ev_instance)){
+        log_e("event_handler_instance_register for IP_EVENT Failed!");
+        return false;
+    }
+    return true;
+}
+
+bool NetworkInterface::started() const {
+    return (getStatusBits() & ESP_NETIF_STARTED_BIT) != 0;
+}
+
+bool NetworkInterface::connected() const {
+    return (getStatusBits() & ESP_NETIF_CONNECTED_BIT) != 0;
+}
+
+bool NetworkInterface::hasIP() const {
+    return (getStatusBits() & ESP_NETIF_HAS_IP_BIT) != 0;
+}
+
+bool NetworkInterface::hasLinkLocalIPv6() const {
+    return (getStatusBits() & ESP_NETIF_HAS_LOCAL_IP6_BIT) != 0;
+}
+
+bool NetworkInterface::hasGlobalIPv6() const {
+    return (getStatusBits() & ESP_NETIF_HAS_GLOBAL_IP6_BIT) != 0;
+}
+
+bool NetworkInterface::enableIPv6(bool en)
+{
+    if (en) {
+        setStatusBits(ESP_NETIF_WANT_IP6_BIT);
+    } else {
+        clearStatusBits(ESP_NETIF_WANT_IP6_BIT);
+    }
+    return true;
+}
+
+bool NetworkInterface::dnsIP(uint8_t dns_no, IPAddress ip)
+{
+    if(_esp_netif == NULL || dns_no > 2){
+        return false;
+    }
+    if(_is_server_if && dns_no > 0){
+        log_e("Server interfaces can have only one DNS server.");
+        return false;
+    }
+    esp_netif_dns_info_t d;
+    // ToDo: can this work with IPv6 addresses?
+    d.ip.type = IPADDR_TYPE_V4;
+    if(static_cast<uint32_t>(ip) != 0){
+        d.ip.u_addr.ip4.addr = static_cast<uint32_t>(ip);
+    } else {
+        d.ip.u_addr.ip4.addr = 0;
+    }
+    if(esp_netif_set_dns_info(_esp_netif, (esp_netif_dns_type_t)dns_no, &d) != ESP_OK){
+        return false;
+    }
+    return true;
+}
+
+bool NetworkInterface::config(IPAddress local_ip, IPAddress gateway, IPAddress subnet, IPAddress dns1, IPAddress dns2, IPAddress dns3)
+{
+    if(_esp_netif == NULL){
+        return false;
+    }
+    esp_err_t err = ESP_OK;
+    esp_netif_ip_info_t info;
+    esp_netif_dns_info_t d1;
+    esp_netif_dns_info_t d2;
+    esp_netif_dns_info_t d3;
+    d1.ip.type = IPADDR_TYPE_V4;
+    d2.ip.type = IPADDR_TYPE_V4;
+    d3.ip.type = IPADDR_TYPE_V4;
+
+    if(static_cast<uint32_t>(local_ip) != 0){
+        info.ip.addr = static_cast<uint32_t>(local_ip);
+        info.gw.addr = static_cast<uint32_t>(gateway);
+        info.netmask.addr = static_cast<uint32_t>(subnet);
+        d1.ip.u_addr.ip4.addr = static_cast<uint32_t>(dns1);
+        d2.ip.u_addr.ip4.addr = static_cast<uint32_t>(dns2);
+        d3.ip.u_addr.ip4.addr = static_cast<uint32_t>(dns3);
+    } else {
+        info.ip.addr = 0;
+        info.gw.addr = 0;
+        info.netmask.addr = 0;
+        d1.ip.u_addr.ip4.addr = 0;
+        d2.ip.u_addr.ip4.addr = 0;
+        d3.ip.u_addr.ip4.addr = 0;
+	}
+
+    if(_is_server_if){
```

Comment:
Things missing from the server config is IPv6, which uses a different approach (RA instead of DHCPS).

I don't think adding IPv6 AP should be part of this change, but a separate thing.  Actually maybe the first thing to add would be IPv6 STA config with a static address, DNS, etc. (also a separate thing).

A long journey, step by step.